src/main/scala/cromwell/engine/CallActor.scala

```diff
@@ -1,46 +1,88 @@
 package cromwell.engine
 
+import java.util.UUID
+import java.util.concurrent.Executors
+
 import akka.actor.{Actor, Props}
 import akka.event.{Logging, LoggingReceive}
 import cromwell.binding._
 import cromwell.binding.values.WdlValue
-import cromwell.engine
-import cromwell.engine.backend.Backend
+import cromwell.engine.backend.{TaskAbortedException, Backend}
 import cromwell.engine.workflow.WorkflowActor
 import cromwell.engine.workflow.WorkflowActor.CallFailed
 
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
-import scala.util.{Failure, Success}
-
+import scala.util.{Try, Failure, Success}
 
 object CallActor {
+  type CallExecutionID = UUID
+  type WorkflowID = UUID
+  type ComboID = (CallExecutionID, WorkflowID)
+
   sealed trait CallActorMessage
-  case class Start(inputs: CallInputs) extends CallActorMessage
-  case class ExecutionFinished(outputs: CallOutputs) extends CallActorMessage
-  case class ExecutionFailed(regret: Throwable) extends CallActorMessage
+  case class Start(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends CallActorMessage
+  case class ProvideCallAbortButton(comboID: ComboID, cancelButton: ()=>Unit) extends CallActorMessage
+  case class AbortCall(workflowId: WorkflowID)
+  case class ExecutionFinished(callExecutionId: UUID, call: Call, outputs: Try[CallOutputs]) extends CallActorMessage
 
-  def props(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor): Props =
-    Props(new CallActor(call, locallyQualifiedInputs, backend, workflowDescriptor))
+  def props(): Props = Props(new CallActor())
 }
 
 /** Actor to manage the execution of a single call. */
-class CallActor(call: Call, locallyQualifiedInputs: Map[String, WdlValue], backend: Backend, workflowDescriptor: WorkflowDescriptor) extends Actor with CromwellActor {
+class CallActor() extends Actor with CromwellActor {
+
+  import CallActor._
 
   type CallOutputs = Map[String, WdlValue]
+  type CancelButton = ()=>Unit
+  val callActorId: UUID = UUID.randomUUID()
+
+  // NB: Should only have one entry but the list is useful for safe mutability.
+  var cancelButtonList: mutable.MutableList[(ComboID, CancelButton)] = new mutable.MutableList[(ComboID, CancelButton)]()
 
   private val log = Logging(context.system, classOf[CallActor])
-  val tag = s"CallActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
+  val tag = s"CallActor [UUID($callActorId)]"
```

Comment:
Following the 'how can we do blocking' discussion, this was preparing for a world where the CallActor==Task link wasn't necessarily 1:1. Hence why log lines now mention the call ID instead of the Actor's tag